### PR TITLE
feat(components): migrate LocationIcon to DeckInfoLabel

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

updates and renames the location icon component to reflect helix/ODD styles

closes PLAT-370, PLAT-408, RQA-2827

Screen Shot 2024-07-30 at 6 34 17 PM
Screen Shot 2024-07-30 at 6 34 22 PM
Screen Shot 2024-07-30 at 6 35 10 PM
Screen Shot 2024-07-30 at 6 36 01 PM
Screen Shot 2024-07-30 at 6 36 22 PM

## Test Plan and Hands on Testing

verified styles and updated unit tests

## Changelog

 - Creates DeckInfoLabel component and migrates LocationIcon

## Review requests

smoke test the various places the label appears, including deck map labels

## Risk assessment

low

### CODE REVIEW

File: components/src/atoms/StyledText/StyledText.tsx [hunk 1]
```diff
@@ -300,12 +308,13 @@ function styleForODDName(name?: ODDStyles): FlattenSimpleInterpolation {
...
 
-// this is some artifact of the way styled-text forwards arguments.
-/* eslint-disable @typescript-eslint/no-unsafe-argument */
-const DesktopStyledText: (props: Props) => JSX.Element = styled(Text)<Props>`
-  ${props => styleForDesktopName(props.desktopStyle)}
+const DesktopStyledText: (props: Props) => JSX.Element = styled(
+  Text
+).withConfig({
+  shouldForwardProp: p => p !== 'oddStyle' && p !== 'desktopStyle',
+})`
```
Comment (brenthagen) [inline components/src/atoms/StyledText/StyledText.tsx:315]: @sfoster1 this un-forwards the non-style props and removes the console warnings `Warning: React does not recognize the oddStyle prop on a DOM element. `

Comment (sfoster1) [reply components/src/atoms/StyledText/StyledText.tsx:315]: Blessed, thank you

**Review by sfoster1 — APPROVED** [review]
Looks good to me, thanks for fixing this!

**Review by smb2268 — APPROVED** [review]
changes lgtm!

**Pull request merged**